tests/security_test.go

```diff
@@ -295,3 +297,96 @@ func runTestCase(tc securityTestCase) error {
 
 	return err
 }
+
+func TestEnvBasedSecret(t *testing.T) {
+	secretName := "env-based-secret"
+	secretKey := "secretKey"
+	jobDescription, err := runSecretTest(secretName, "", secretKey)
+
+	if err != nil {
+		t.Error(err.Error())
+	}
+
+	if strings.Contains(jobDescription, fmt.Sprintf("set to the key '%s' in secret '%s'", secretKey, secretName)) {
+		log.Infof("Successfully set environment variable to the key '%s' in secret '%s'", secretKey, secretName)
+	} else {
+		t.Errorf("Unnable to set environment variable to the key '%s' in secret '%s'", secretKey, secretName)
+	}
+}
+
+func TestFileBasedSecrets(t *testing.T) {
```

Comment:
These two methods(`TestEnvBasedSecret` and `TestFileBasedSecrets`)  are looking very similar. maybe it makes sense to extract the logic into another one method and invoke it with different parameters depending on test case.  
WDYT, @farhan5900 ?

Reply:
Right these two methods are almost same, that's why we have `runSecretTest` method containing all common logic. We can extract remaining logic as well. I will try to do that.